modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/GridCacheDatabaseSharedManager.java

```diff
@@ -4896,9 +4836,10 @@ public void awaitPendingTasksFinished() throws IgniteCheckedException {
      * @param pagesArr Pages array.
      * @param cmp Cmp.
      */
-    private static void parallelSortInIsolatedPool(
+    private static ExecutorService parallelSortInIsolatedPool(
         FullPageId[] pagesArr,
-        Comparator<FullPageId> cmp
+        Comparator<FullPageId> cmp,
+        ExecutorService pool
```

Comment:
Please use ForkJoinPool as parameter type. Not every executor service is suitable for the job.